rust/core-lib/src/watcher.rs

```diff
@@ -34,12 +34,12 @@
 //! - We are integrated with the xi_rpc runloop; events are queued as
 //! they arrive, and an idle task is scheduled.
 
-use notify::{watcher, DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
+use crossbeam::unbounded;
+use notify::{event::*, watcher, RecommendedWatcher, RecursiveMode, Watcher};
```

Comment:
I have noticed `use` statements within methods. Is it because we dont want them to be loaded/linked in case we are not using notify?

Reply:
> Is it because we dont want them to be loaded/linked in case we are not using notify?

Unlikely, you can put `#[cfg(feature = "notify")]` above normal use statements too. Not quite sure what that is about

Reply:
Maybe it's due to naming conflicts

Reply:
@Cogitri 
Im not sure either. I have mostly seen scoped `use` statements in testcases. By the way did you get a chance to look at https://github.com/xi-editor/xi-editor/issues/1158#issuecomment-507067284? I can provide more details if required.

Reply:
Commented on that.